include/path/SwervePathBuilder.h

```diff
@@ -0,0 +1,249 @@
+// Copyright (c) TrajoptLib contributors
+
+#pragma once
+
+#include <vector>
+
+#include "drivetrain/SwerveDrivetrain.h"
+#include "obstacle/Bumpers.h"
+#include "obstacle/Obstacle.h"
+#include "path/InitialGuessPoint.h"
+#include "path/Path.h"
+#include "set/IntervalSet1d.h"
+#include "set/Set2d.h"
+
+namespace trajopt {
+
+/**
+ * Builds a swerve path using information about how the robot
+ * must travel through a series of waypoints. This path can be converted
+ * to a trajectory using OptimalTrajectoryGenerator.
+ */
+class TRAJOPT_DLLEXPORT SwervePathBuilder {
+ public:
+  /**
+   * Get the SwervePath being constructed
+   *
+   * @return the path
+   */
+  const SwervePath& GetPath() const;
+
+  /**
+   * Set the Drivetrain object
+   *
+   * @param drivetrain the new drivetrain
+   */
+  void SetDrivetrain(SwerveDrivetrain drivetrain);
+
+  /**
+   * Create a pose waypoint constraint on the waypoint at the provided
+   * index, and add an initial guess with the same pose This specifies that the
+   * position and heading of the robot at the waypoint must be fixed at the
+   * values provided.
+   *
+   * @param idx index of the pose waypoint
+   * @param x the x
+   * @param y the y
+   * @param heading the heading
+   */
+  void PoseWpt(size_t idx, double x, double y, double heading);
```

Comment:
I'd use the full name `Waypoint` instead of `Wpt` for clarity.

Reply:
I’ve gone back and forth on this, but I really think the abbreviation is helpful because otherwise some of the names get really long. I can document that in the class docs.